### DISCUSSION TITLE
Support for `isFocusWithin` in `GridList` `Item`

### DISCUSSION THREAD

**Discussion by jaknas** [adobe/react-spectrum#4849] [General]
Hello 👋 

Currently exposed [render props and css selectors](https://react-spectrum.adobe.com/react-aria/GridList.html#item-1) are very useful for styling. 

From time to time, I want to add some styles to Item when it's interacted with (e.g add box-shadow). I can already use `data-hovered` etc.. to do that, but for cases when user moves with focus inside an `Item` (e.g moving between interactive elements inside) I'm missing a selector that would apply some styles to `Item`.

I was thinking that `data-focus-within` could be added to `Item` whenever focus is inside this Item's descendants. 

So my question is if there's any reason why it wasn't added like the other focus handlers, and whether you'd be open to adding it.

**Comment by jaknas** [comment]
Just wanted to bump this, as I've been struggling with styling this state also when using `<Row/>` component in a `<Table/>`, where I'd like to maintain Row hover state when focus is being moved inside the row children

**Comment by reidbarber** [comment]
Have you seen the [Group](https://react-spectrum.adobe.com/react-aria/Group.html) element? It may give you some more freedom with focus/hover states. Not sure it will help with things like tables which require specific elements as children.

> I was thinking that data-focus-within could be added to Item whenever focus is inside this Item's descendants.

I think this is probably a good idea. Feel free to open an issue so there can be some more discussion around it. 

> Just wanted to bump this, as I've been struggling with styling this state also when using `<Row/>` component in a `<Table/>`, where I'd like to maintain Row hover state when focus is being moved inside the row children.

So if I'm understanding this correctly, you would like to have the same style when the row is hovered and when focus is within a row? But you don't know at the row level if focus is inside the row or not. This seems related to the above and I would say lets include that in the issue. These definitely seem like valid use cases. Thanks for raising the discussion!

**Comment by jaknas** [comment]
> So if I'm understanding this correctly, you would like to have the same style when the row is hovered and when focus is within a row? But you don't know at the row level if focus is inside the row or not. This seems related to the above and I would say lets include that in the issue. These definitely seem like valid use cases. Thanks for raising the discussion!

Yes, this is exactly what I'm after. I opened [this issue](https://github.com/adobe/react-spectrum/issues/5258) as you suggested. Thanks for answering!
